api/types/installers/installer.sh.tmpl

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+( flock -n 9 || exit 1;
+  if test -f /usr/local/bin/teleport; then
+	  exit 0
+  fi
+
+  distro_id="$(awk -F= '$1 == "ID" { print tolower($2) }' /etc/os-release | xargs echo)"
+  if [ "$distro_id" = "debian" ] || [ "$distro_id" = "ubuntu" ]; then
+	  sudo curl https://deb.releases.teleport.dev/teleport-pubkey.asc \
+		   -o /usr/share/keyrings/teleport-archive-keyring.asc
+	  source /etc/os-release
+	  echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc]  https://apt.releases.teleport.dev/${ID?} ${VERSION_CODENAME?} stable/{{ .MajorVersion }}" \
+		  | sudo tee /etc/apt/sources.list.d/teleport.list > /dev/null
+	  sudo apt-get update
+	  sudo apt-get install teleport jq
+  elif [ "$distro_id" = "amzn" ] || [ "$distro_id" = "rhel" ]; then
+	  source /etc/os-release
+	  sudo yum-config-manager --add-repo \
+		   $(rpm --eval "https://yum.releases.teleport.dev/$ID/$VERSION_ID/Teleport/%{_arch}/stable/{{ .MajorVersion }}/teleport.repo")
+	  sudo yum install -y teleport jq
+  fi
+
+  INSTANCE_INFO="$(curl http://169.254.169.254/latest/dynamic/instance-identity/document)"
+
+  ACCOUNT_ID="$(echo "$INSTANCE_INFO" | jq -r .accountId)"
+  INSTANCE_ID="$(echo "$INSTANCE_INFO" | jq -r .instanceId)"
+
+  # generate teleport ssh config
+  # token is read as a parameter from the AWS ssm script run and
+  # passed as the first argument to the script
+  /usr/local/bin/teleport node configure \
+						  --auth-server="{{ .PublicProxyAddr }}" \
+						  --join-method=iam \
+						  --token="$1" \
+						  --labels="teleport.dev/instance-id=${INSTANCE_ID},teleport.dev/account-id=${ACCOUNT_ID}" \
+						  --node-name="${ACCOUNT_ID}-${INSTANCE_ID}" | sudo tee "/etc/teleport.yaml"
```

Comment:
Why do we need to set the node name explicitly? What if we omit this flag here?

Reply:
Also, I think that the "configure" command accepts `-o=file` flag that will write it to /etc/teleport.yaml so you don't need to pipe to tee.

Reply:
Hmm I see, I think with auto-discovery we don't expect the config to exist so overwriting is probably fine. Let's keep your approach for now.